libultraship/libultraship/GlobalCtx2.cpp

```diff
@@ -29,6 +32,23 @@ namespace Ship {
         return GetInstance();
     }
 
+    std::string GlobalCtx2::GetAppDirectoryPath() {
+        #ifdef __APPLE__
+            char *folderPath = NULL;
+            FolderManager folderManager;
+            auto appSupportDirectory = (char *)folderManager.pathForDirectory(NSApplicationSupportDirectory, NSUserDomainMask);
+
+            char buf[100];
+            strcpy(buf, appSupportDirectory);
+            strcat(buf, "/com.shipofharkinian.soh");
+
+            return buf;
```

Comment:
this is a stack use after return, should be heap allocated

Reply:
Yeah, this came from my lack of knowledge in C++. I initially wanted to return `appSupportDirectory + "/com.shipofharkinian.soh";` but got an error that you cannot use + with `'char *' and 'const char[25]'` and read online this was the best way to add these 😅

regarding the stack vs heap topic - do you mean I should write this as?:
```c++
char* buf = new char[100];
strcpy(buf, appSupportDirectory);
strcat(buf, "/com.shipofharkinian.soh");

return std::string(buf);
```